File: src/main/java/com/github/nsc/de/shake/interpreter/InterpreterError.java
package com.github.nsc.de.shake.interpreter;

import com.github.nsc.de.shake.util.CompilerError;
import com.github.nsc.de.shake.util.characterinput.position.Position;
import com.github.nsc.de.shake.util.characterinput.position.PositionMap;

public class InterpreterError extends CompilerError {

    public InterpreterError(String message, String details, Position start, Position end, Throwable cause) {
        super(message, "InterpretationError", details, start, end, cause);
    }

    public InterpreterError(String message, String details, PositionMap map, int start, int end, Throwable cause) {
        super(message, "InterpretationError", details, map, start, end, cause);
    }

    public InterpreterError(String message, String details, PositionMap map, int position, Throwable cause) {
        this(message, details, map, position, position, cause);
    }

    public InterpreterError(String details, PositionMap map, int start, int end, Throwable cause) {
        this("Error occurred in Interpreter", details, map, start, end, cause);
    }

    public InterpreterError(String details, PositionMap map, int position, Throwable cause) {
        this("Error occurred in Interpreter", details, map, position, cause);
    }

    public InterpreterError(String message, String details, Position start, Position end) {
        super(message, "InterpretationError", details, start, end);
    }

    public InterpreterError(String message, String details, PositionMap map, int start, int end) {
        super(message, "InterpretationError", details, map, start, end);
    }

    public InterpreterError(String message, String details, PositionMap map, int position) {
        this(message, details, map, position, position);
    }

    public InterpreterError(String details, PositionMap map, int start, int end) {
        this("Error occurred in Interpreter", details, map, start, end);
    }

    public InterpreterError(String details, PositionMap map, int position) {
        this("Error occurred in Interpreter", details, map, position);
    }

}

File: interpreter/src/commonMain/kotlin/com/github/nsc/de/shake/interpreter/InterpreterError.kt
package com.github.nsc.de.shake.interpreter

import com.github.nsc.de.shake.util.CompilerError
import com.github.nsc.de.shake.util.characterinput.position.Position
import com.github.nsc.de.shake.util.characterinput.position.PositionMap

@Suppress("unused")
class InterpreterError : CompilerError {
    constructor(message: String?, details: String?, start: Position?, end: Position?, cause: Throwable?) :
            super(message!!, "InterpretationError", details!!, start!!, end!!, cause!!)

    constructor(message: String?, details: String?, map: PositionMap?, start: Int, end: Int, cause: Throwable?) :
            super(message!!, "InterpretationError", details!!, map!!, start, end, cause!!)

    constructor(message: String?, details: String?, map: PositionMap?, position: Int, cause: Throwable?) :
            this(message, details, map, position, position, cause)

    constructor(details: String?, map: PositionMap?, start: Int, end: Int, cause: Throwable?) :
            this("Error occurred in Interpreter", details, map, start, end, cause)

    constructor(details: String?, map: PositionMap?, position: Int, cause: Throwable? ) :
            this("Error occurred in Interpreter", details, map, position, cause)

    constructor(message: String?, details: String?, start: Position?, end: Position?) :
            super(message!!, "InterpretationError", details!!, start!!, end!!)

    constructor(message: String?, details: String?, map: PositionMap?, start: Int, end: Int) :
            super(message!!, "InterpretationError", details!!, map!!, start, end)

    constructor(message: String?, details: String?, map: PositionMap?, position: Int) :
            this(message, details, map, position, position)

    constructor(details: String?, map: PositionMap?, start: Int, end: Int) :
            this("Error occurred in Interpreter", details, map, start, end)

    constructor(details: String?, map: PositionMap?, position: Int) :
            this("Error occurred in Interpreter", details, map, position )
}